funding/manager.go

```diff
@@ -1039,13 +1056,85 @@ func (f *Manager) stateStep(channel *channeldb.OpenChannel,
 	return fmt.Errorf("undefined channelState: %v", channelState)
 }
 
+// waitForFundingSpend waits for any of the inputs to the provided funding
+// transaction to be spent, and returns the txid of the spending transaction.
+// This is used to detect the case where the funding tx is double spent and
+// rendered invalid.
+func (f *Manager) waitForFundingSpend(fundingTx *wire.MsgTx,
+	fndBroadcastHeight uint32) (*chainhash.Hash, error) {
+
+	quit := make(chan struct{})
```

Comment:
I don't think we need _two_ quit channels here. Instead, we should be able to rely on the funding manager level quit channel. Otherwise, we should pass in a context here, so we can cancel the funding spend goroutine if we end up abandoning the funding attempt some other way.

Reply:
👍